java-vtl-script/src/test/java/no/ssb/vtl/script/support/DatasetCloseWatcher.java

```diff
@@ -0,0 +1,90 @@
+package no.ssb.vtl.script.support;
+
+import com.google.common.collect.ForwardingObject;
+import no.ssb.vtl.model.DataPoint;
+import no.ssb.vtl.model.DataStructure;
+import no.ssb.vtl.model.Dataset;
+import no.ssb.vtl.model.Order;
+
+import java.util.Map;
+import java.util.Optional;
+import java.util.Set;
+import java.util.stream.Stream;
+
+public abstract class DatasetCloseWatcher extends ForwardingObject implements Dataset {
+
+    private int counter = 0;
+
+    private DatasetCloseWatcher() {
+        // disable
+    }
+
+    protected abstract Dataset delegate();
+
+    public boolean allStreamWereClosed() {
+        return counter == 0;
+    }
+
+    public static DatasetCloseWatcher wrap(Dataset dataset) {
+        return new DatasetCloseWatcher() {
+
+            @Override
+            protected Dataset delegate() {
+                return dataset;
+            }
+        };
+    }
+
+    @Override
+    public Stream<DataPoint> getData() {
+        return wrap(delegate().getData());
+    }
+
+    private void increase() {
+        this.counter++;
+    }
+
+    private void decrease() {
+        this.counter--;
+    }
+
+    private Stream<DataPoint> wrap(Stream<DataPoint> stream) {
+        this.increase();
```

Comment:
I'm wondering if increasing and decreasing the counter should be thread-safe or not.

Reply:
Thought of it, I don't think it is necessary.